src/main/java/jsr223/docker/compose/DockerComposeScriptEngine.java

```diff
@@ -124,26 +146,38 @@ private String replaceVariables(String script, Map<String, String> variables) {
 
     /**
      * Adds all bindings which are from type @String to the environment map. All other bindings are printed
-     * with toString() to std.err with an error message.
+     * with toString() to log file.
      * @param bindings Bindings which will be read and added to environment.
-     * @param environment Map<String,String> which will get all Entry<String,String> from he @Bindings
+     * @param environment Map<String,String> which will get all Entry<String,String> from the @Bindings
      */
     private void addBindingToStringMap(@NotNull Bindings bindings, @NotNull Map<String, String> environment) {
         for(Map.Entry<String, Object> entry: bindings.entrySet()) {
             if (entry.getValue() instanceof String) {
-                environment.put(entry.getKey(),(String) entry.getValue());
-                System.out.println("Added binding: "+entry.getKey()+":"+entry.getValue().toString());
+                addEntryToEnvironmentWhichIsAPureString(environment, entry);
             } else { // Go through maps and add String String values to the environment map.
-                if (entry.getValue() instanceof Map<?, ?>)
-                    for (Map.Entry<?, ?> mapEntry :  ((Map<?,?>) entry.getValue()).entrySet()) {
-                        if (mapEntry.getValue() instanceof String && mapEntry.getKey() instanceof String) {
-                            environment.put((String) mapEntry.getKey(), (String) mapEntry.getValue());
-                            System.out.println("Added binding: "+mapEntry.getKey()+":"+mapEntry.getValue().toString());
-                        }
-                    }
-                else {
-                    System.err.println("Ignored binding: " + entry.getKey() + "(" + entry.getKey().getClass().getName() + "):" + entry.getValue().toString() + "(" + entry.getValue().getClass().getName() + ")");
-                }
+                AddEntryToEnvironmentOtherThanPureStrings(environment, entry);
+            }
+        }
+    }
+
+    private void addEntryToEnvironmentWhichIsAPureString(@NotNull Map<String, String> environment, Map.Entry<String, Object> entry) {
+        environment.put(entry.getKey(),(String) entry.getValue());
+        log.debug("Added binding: "+entry.getKey()+":"+entry.getValue().toString());
+    }
+
+    private void AddEntryToEnvironmentOtherThanPureStrings(@NotNull Map<String, String> environment, Map.Entry<String, Object> entry) {
+        if (entry.getValue() instanceof Map<?, ?>)
+            addEntryToEnvironmentWhichIsAMapContainingStrings(environment, entry);
+        else {
+            log.warn("Ignored binding: " + entry.getKey() + "(" + entry.getKey().getClass().getName() + "):" + entry.getValue().toString() + "(" + entry.getValue().getClass().getName() + ")");
```

Comment:
Could extract `entry.getKey().getClass().getName()` to factorize code and improve readability.